pkg/pool-manager/pool.go

```diff
@@ -405,3 +401,78 @@ func (p *PoolManager) getOptMode(mutatingWebhookConfigName, webhookName string)
 
 	return "", fmt.Errorf("No Opt mode defined for webhook %s in mutatingWebhookConfig %s", webhookName, mutatingWebhookConfigName)
 }
+
+// UpdateRanges atomically updates the MAC address ranges for allocation
+// This method will be used for dynamic range updates without pod restart
+func (p *PoolManager) UpdateRanges(newStart, newEnd net.HardwareAddr) error {
+	// Validate the new ranges
+	if err := checkRange(newStart, newEnd); err != nil {
+		return err
+	}
+
+	if err := checkCast(newStart); err != nil {
+		return fmt.Errorf("invalid range start: %v", err)
+	}
+
+	if err := checkCast(newEnd); err != nil {
+		return fmt.Errorf("invalid range end: %v", err)
+	}
+
+	// Update ranges atomically
+	p.rangeMutex.Lock()
+	defer p.rangeMutex.Unlock()
+
+	oldStart := p.rangeStart
+	oldEnd := p.rangeEnd
+
+	p.rangeStart = newStart
+	p.rangeEnd = newEnd
+
+	// Reset current MAC to somewhere in the new range
+	newCurrentMac, err := generateRandomMac(newStart, newEnd)
+	if err != nil {
+		// Rollback on failure
+		p.rangeStart = oldStart
+		p.rangeEnd = oldEnd
+		return fmt.Errorf("failed to generate new current MAC: %v", err)
+	}
+	p.currentMac = newCurrentMac
+
+	log.Info("Successfully updated MAC allocation ranges",
+		"oldStart", oldStart.String(), "oldEnd", oldEnd.String(),
+		"newStart", newStart.String(), "newEnd", newEnd.String(),
+		"newCurrentMac", newCurrentMac.String())
+
+	return nil
+}
+
+// GetCurrentRanges returns the current MAC address ranges
+func (p *PoolManager) GetCurrentRanges() (start, end net.HardwareAddr) {
+	p.rangeMutex.RLock()
+	defer p.rangeMutex.RUnlock()
+
+	// Return copies to avoid external modification
+	start = make(net.HardwareAddr, len(p.rangeStart))
+	end = make(net.HardwareAddr, len(p.rangeEnd))
+	copy(start, p.rangeStart)
+	copy(end, p.rangeEnd)
+
+	return start, end
+}
+
+// getCurrentMAC returns the current MAC address with proper locking
+func (p *PoolManager) getCurrentMAC() net.HardwareAddr {
+	p.rangeMutex.RLock()
+	defer p.rangeMutex.RUnlock()
+
+	// Return a copy to avoid external modification
+	currentMac := make(net.HardwareAddr, len(p.currentMac))
+	copy(currentMac, p.currentMac)
+
```

Comment:
Ditto: maybe we just need to return the string representation

Reply:
yep..